cmd/tailscale/cli/serve.go

```diff
@@ -7,46 +7,105 @@ package cli
-		FlagSet:    e.newFlags("serve", func(fs *flag.FlagSet) {}),
+		Name:      "serve",
+		ShortHelp: "[ALPHA] Serve from your Tailscale node",
+		ShortUsage: strings.TrimSpace(`
+  serve [flags] <mount-point> {proxy|path|text} <arg>
+  serve [flags] <sub-command> [sub-flags] <args>`),
+		LongHelp: strings.TrimSpace(`
+*** ALPHA; all of this is subject to change ***
+
+The 'tailscale serve' set of commands allows you to serve
+content and local servers from your Tailscale node to
+your tailnet. 
+
+You can also choose to enable the Tailscale Funnel with:
+'tailscale serve funnel on'.
+
+This allows your services to be accessed over the internet,
+securely using TLS and of course, Tailscale.
+
+EXAMPLES
+  - To proxy requests to a web server at 127.0.0.1:3000:
+    $ tailscale serve / proxy 3000
+
+  - To serve a single file or a directory of files:
+    $ tailscale serve / path /home/alice/blog/index.html
+    $ tailscale serve /images/ path /home/alice/blog/images
+
+  - To serve simple static text:
+    $ tailscale serve / text "Hello, world!"
+`),
+		Exec: e.runServe,
+		FlagSet: e.newFlags("serve", func(fs *flag.FlagSet) {
+			fs.BoolVar(&e.remove, "remove", false, "remove an existing serve config")
+			fs.UintVar(&e.servePort, "serve-port", 443, "port to serve on (443, 8443 or 10000)")
+		}),
+		UsageFunc: usageFunc,
 		Subcommands: []*ffcli.Command{
 			{
-				Name:      "show-config",
-				Exec:      e.runServeShowConfig,
-				ShortHelp: "show current serve config",
+				Name:      "status",
+				Exec:      e.runServeStatus,
+				ShortHelp: "show current serve status",
+				FlagSet: e.newFlags("serve-status", func(fs *flag.FlagSet) {
+					fs.BoolVar(&e.json, "json", false, "output JSON")
+				}),
+				UsageFunc: usageFunc,
 			},
 			{
 				Name:      "tcp",
 				Exec:      e.runServeTCP,
 				ShortHelp: "add or remove a TCP port forward",
+				LongHelp: strings.Join([]string{
+					"EXAMPLES",
+					"  - Forward TLS over TCP to a local TCP server on port 5432:",
+					"    $ tailscale serve tcp 5432",
+					"",
+					"  - Forward raw, TLS-terminated TCP packets to a local TCP server on port 5432:",
+					"    $ tailscale serve --terminate-tls tcp 5432",
+				}, "\n"),
 				FlagSet: e.newFlags("serve-tcp", func(fs *flag.FlagSet) {
 					fs.BoolVar(&e.terminateTLS, "terminate-tls", false, "terminate TLS before forwarding TCP connection")
+					fs.UintVar(&e.servePort, "serve-port", 443, "port to serve on (443, 8443 or 10000)")
 				}),
+				UsageFunc: usageFunc,
 			},
 			{
-				Name:      "ingress",
-				Exec:      e.runServeIngress,
-				ShortHelp: "enable or disable ingress",
-				FlagSet:   e.newFlags("serve-ingress", func(fs *flag.FlagSet) {}),
+				Name:      "funnel",
+				Exec:      e.runServeFunnel,
+				ShortHelp: "turn on or off the Tailscale Funnel",
+				FlagSet: e.newFlags("serve-funnel", func(fs *flag.FlagSet) {
+					fs.UintVar(&e.servePort, "serve-port", 443, "port to serve on (443, 8443 or 10000)")
```

Comment:
just "port" for flag? and this is the port to turn on or off. 

Note (in LongHelp?) that it doesn't turn off serving on the tailnet; only the internet.

Reply:
I actually think it was a mistake I put `serve-port` flag here and in TCP as well.

`--serve-port` scopes the entire command regardless of sub-command. It also maps existing servers 1:1 with the funnel port.

```console
# serve defaults to --serve-port=443
# serve listens locally on tailscale-ip;443 
# funnel on will funnel into tailscale-ip:443 be default
$ tailscale serve / proxy 3000
$ tailscale serve funnel on

# specifying --serve-port=10000 scopes the command
# to listen locally on tailscale-ip:10000
# funnel with explicit flag will funnel into tailscale-ip:10000
$ tailscale serve --serve-port=10000 tcp 5432
$ tailscale serve --serve-port=10000 funnel on
```

Reply:
I've removed `--serve-port` from sub-commands, leaving only the top level. LMK